x-pack/legacy/plugins/siem/public/components/generic_downloader/index.tsx

```diff
@@ -6,18 +6,28 @@
 
 import React, { useEffect, useRef } from 'react';
 import styled from 'styled-components';
-import { isFunction } from 'lodash/fp';
-import { exportRules } from '../../../../../containers/detection_engine/rules';
-import { useStateToaster, errorToToaster } from '../../../../../components/toasters';
+import { isFunction, isNil } from 'lodash/fp';
 import * as i18n from './translations';
+import { ExportTimelineIds } from '../open_timeline/open_timeline';
+import { exportRules, ExportDocumentsProps } from '../../containers/detection_engine/rules';
+import { useStateToaster, errorToToaster } from '../toasters';
 
 const InvisibleAnchor = styled.a`
   display: none;
 `;
 
+export type ExportSelectedData = ({
+  excludeExportDetails,
+  filename,
+  ids,
+  signal,
+}: ExportDocumentsProps) => Promise<Blob>;
+
 export interface RuleDownloaderProps {
   filename: string;
+  ids?: ExportTimelineIds[];
```

Comment:
maybe we can do something like that to avoid adding more props when we will need to export more objects like case/ml etc...
```
interface DownloadRules {
  type: 'rules';
  ids: string;
}
interface DownloadTimelines {
  type: 'timelines';
  ids: ExportTimelineIds;
}
```
and then
```
export interface RuleDownloaderProps {
  filename: string;
  download: DownloadRules | DownloadTimelines;
  etc...
}
```

Reply:
After a zoom discussion, we decided to use the same type of Ids, which is string[], for exporting timeline.